lib/models/connection_status.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:connectivity/connectivity.dart';
+import 'package:flutter/material.dart';
+
+class ConnectionStatus extends ChangeNotifier {
+  final Connectivity _connectivity = Connectivity();
+  bool hasConnection;
+  bool previousConnection;
+  bool shouldShowSnackbar = false;
+
+  ConnectionStatus() {
+    _initializeConnection();
+
+    _connectivity.onConnectivityChanged.listen(_connectionChange);
+  }
+
+  void _connectionChange(ConnectivityResult result) {
+    checkConnection(result);
+  }
+
+  checkConnection(ConnectivityResult status) async {
+    previousConnection = hasConnection;
+    if (status == ConnectivityResult.mobile ||
+        status == ConnectivityResult.wifi) {
+      try {
+        final lookup = await InternetAddress.lookup('google.com');
```

Comment:
Why not make this URL a class variable so that we can change it to our own server IP when we host it? We would not lookup `www.google.com` cause we need to ping our own server in ideal case

Reply:
![status](https://user-images.githubusercontent.com/37744935/82635708-70953500-9c1e-11ea-891d-ecf5b5ce1b83.gif)

Actually this snackbar idea is creating some problem, according to me we cant create multiple snackbar which would be the case if we use snackbar in each screen as some of the screens are not disposed yet, we just push our current screen's widget above the previous one. 

So this notification or a modal should definitely be a better idea than this.

Reply:
About pinging our own server , I was thinking could i reuse websocket.dart . If we can  , could you help me in that , cause we can't use socket's Ip in that function and could not find any function that could ping a socket server